lib/traject/config/sirsi_config.rb

```diff
@@ -312,6 +395,35 @@ def extract_sortable_title(fields, record)
                                          record)
 end
 
+to_field 'author_struct' do |record, accumulator|
+  struct = {}
+  struct[:creator] = linked_author_struct(record, '100')
+  struct[:corporate_author] = linked_author_struct(record, '110')
+  struct[:meeting] = linked_author_struct(record, '111')
+  struct.reject! { |_k, v| v.empty? }
+
+  accumulator << struct unless struct.empty?
+end
+
+def linked_author_struct(record, tag)
+  record.fields(tag).map do |field|
```

Comment:
Ugh; it looks like we'll have to implement the Searchworks MarcField `run_preprocessors` steps too..